docs/source/reference/jinja.rst

```diff
@@ -1,101 +1,87 @@
-for Jinja specific documentation.
+|st2| uses `Jinja <http://jinja.pocoo.org/>`_ extensively for templating. Jinja
+allows you to manipulate parameter values in |st2| by allowing you to refer to
+other parameters, applying filters or refer to system specific constructs (like
+datastore access). This document is here to help you with Jinja in the context
+of |st2|. Please refer to the `Jinja docs <http://jinja.pocoo.org/docs/>`_ for
+Jinja-focused information.
 
-For brevity, the jinja patterns are only documented. Usage of these patterns inside YAML is subject to understanding the YAML for each content type (sensors, triggers, rules, action and workflows) and their syntax.
+.. _applying-filters-with-jinja:
 
-Accessing datastore items with Jinja
+Referencing Datastore Keys in Jinja
 ------------------------------------
 
 You can use ``{{st2kv.system.foo}}`` to access key ``foo`` from datastore. Note that until
 v2.1, the expression to access key ``foo`` from datastore used to be ``{{system.foo}}``
-but is now deprecated.
+but is now deprecated, and the leading ``st2kv.`` namespace is required.
 
-.. _applying-filters-with-jinja:
+.. _jinja-jinja-filters:
 
-Applying filters with Jinja
+Applying Filters with Jinja
 ----------------------------
 
-Aside from `standard filters <http://jinja.pocoo.org/docs/dev/
-templates/#builtin-filters>`_ available in Jinja, |st2| supports custom filters
-as well, see :ref:`jinja-jinja-filters`. To use a filter ``my_filter`` on ``foo``, simply do
-``{{foo | my_filter}}``. Please pay attention to data type and available filters
-for each data type. Since Jinja is a text templating language, all your input is
-converted to text and then manipulations happen on them. The necessary casting at
-the end is done by |st2| based on information you provide in YAML (for example,
+To use a filter ``my_filter`` on ``foo``, you use the pipe operator, like so: ``{{foo | my_filter}}``.
+Please pay attention to data type and available filters for each data type.
+Since Jinja is a text templating language, all your input is converted to text
+and then manipulations happen on that value. The necessary casting at the end is
+done by |st2| based on information you provide in YAML (for example,
 ``type`` field in action parameters). The casting is a best effort casting.
 
-The current supported filters are grouped into following categories.
+|st2| supports `Jinja variable templating <http://jinja.pocoo.org/docs/dev/templates/#variables>`__
+in Rules, Action Chains and Actions etc. Jinja templates support
+`filters <http://jinja.pocoo.org/docs/dev/templates/#list-of-builtin-filters>`__
+to allow some advanced capabilities in working with variables.
+
+.. _referencing-datastore-keys-in-jinja:
+
+Custom Jinja Filters
+--------------------
+
+In addition to `standard filters <http://jinja.pocoo.org/docs/dev/
+templates/#builtin-filters>`_ available in Jinja, |st2| supports custom filters
+as well.
```

Comment:
"In addition to...supports custom filters as well" - feels like we don't need "as well" at the end, as it's redundant?